### PR TITLE
create tests for mp-b2b-abstract-product-image-sets

### PR DESCRIPTION
**Pull request by m-alama**

### CODE REVIEW

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+    
+Get_abstract_image_sets_with_1_concrete
+    When I send a GET request:    /abstract-products/${abstract_available_with_stock_and_never_out_of_stock}/abstract-product-image-sets
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:15]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    And Response body parameter should be:    [data][0][id]    ${abstract_available_with_stock_and_never_out_of_stock}
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:16]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    And Response body has correct self link
+
+
+Get_abstract_product_with_1_concerete_with_include_abstract_product_image_sets
+    When I send a GET request:    /abstract-products/${abstract_available_with_stock_and_never_out_of_stock}?include=abstract-product-image-sets
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response should contain the array of a certain size:    [included]    1
+    And Response should contain the array of a certain size:    [data][relationships]    1
+    And Response body parameter should not be EMPTY:    [data][relationships]
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:44]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    And Response body parameter should not be EMPTY:    [data][relationships][abstract-product-image-sets]
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-image-sets][data]    type
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-image-sets][data]    id
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:47]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    And Response body parameter should not be EMPTY:    [included]
+    And Each array element of array in response should contain property:    [included]    type
+    And Each array element of array in response should contain property:    [included]    id
+    And Each array element of array in response should contain property:    [included]    attributes
+    And Each array element of array in response should contain property:    [included]    links
+    And Each array element of array in response should contain nested property:    [included]    [links]    self
+    And Each array element of array in response should contain property with value:    [included]    type    abstract-product-image-sets
+    And Each array element of array in response should contain nested property:    [included]    attributes    imageSets
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:55]: this line test type for every element is abstract-product-image-sets
and the above lines test the attributes object it self

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/negative.robot [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+Suite Setup       SuiteSetup
+Test Setup        TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+    
+Get_abstract_image_stes_by_concrete_SKU
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/negative.robot:11]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    And Response reason should be:    OK
+    And Response should contain the array of a certain size:    [included]    1
+    And Response should contain the array of a certain size:    [data][relationships]    1
+    And Response body parameter should not be EMPTY:    [data][relationships]
+    And Response body parameter should not be EMPTY:    [data][relationships][abstract-product-image-sets]
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-image-sets][data]    type
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-image-sets][data]    id
+    And Response body parameter should not be EMPTY:    [included]
+    And Each array element of array in response should contain property:    [included]    type
+    And Each array element of array in response should contain property:    [included]    id
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:50]: Done above

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    And Response body parameter should be:    [data][0][id]    ${abstract_available_product_with_3_concretes}
+    And Response should contain the array of a certain size:    [data][0][attributes][imageSets]   1
+    And Response should contain the array of a certain size:    [data][0][attributes][imageSets][0][images]    1
+    And Response body parameter should not be EMPTY:   [data][0][attributes][imageSets][0][images][0][externalUrlLarge]
+    And Response body parameter should not be EMPTY:    [data][0][attributes][imageSets][0][images][0][externalUrlSmall]
+    And Response body parameter should be:    [data][0][attributes][imageSets][0][name]    default
+    And Response body has correct self link
+
+
+Get_abstract_product_with_1_concerete_with_include_abstract_product_image_sets
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:38]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    And Each array element of array in response should contain property with value:    [included]    type    abstract-product-image-sets
+    And Each array element of array in response should contain nested property:    [included]    attributes    imageSets
+    And Each array element of array in response should contain nested property:    [included]    [attributes][imageSets]    name
+    And Each array element of array in response should contain nested property:    [included]    [attributes][imageSets]    images
+    And Each array element of array in response should contain nested property:    [included]    [attributes][imageSets][0][images]    externalUrlLarge
+    And Each array element of array in response should contain nested property:    [included]    [attributes][imageSets][0][images]    externalUrlSmall
+    And Response body has correct self link internal
+
+
+Get_abstract_product_with_3_concretes_with_include_abstract_product_image_sets
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:63]: I can't find the typo

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+Get_abstract_product_with_1_concrete_with_include_abstract_product_image_sets
+    When I send a GET request:    /abstract-products/${abstract_available_with_stock_and_never_out_of_stock}?include=abstract-product-image-sets
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response should contain the array of a certain size:    [included]    1
+    And Response should contain the array of a certain size:    [data][relationships]    1
+    And Response body parameter should not be EMPTY:    [data][relationships][abstract-product-image-sets]
+    And Response body parameter should be:    [data][relationships][abstract-product-image-sets][data][0][type]    abstract-product-image-sets
+    And Response body parameter should be:    [data][relationships][abstract-product-image-sets][data][0][id]    ${abstract_available_with_stock_and_never_out_of_stock}
+    And Response body parameter should not be EMPTY:    [included]
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:46]: Also we don't need to check empty included if we check included parameters below (we expect that it's not empty)

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    When I send a GET request:    /abstract-products/${abstract_available_product_with_3_concretes}?include=abstract-product-image-sets
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response should contain the array of a certain size:    [included]    1
+    And Response should contain the array of a certain size:    [data][relationships]    1
+    And Response body parameter should not be EMPTY:    [data][relationships]
+    And Response body parameter should not be EMPTY:    [data][relationships][abstract-product-image-sets]
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-image-sets][data]    type
+    And Each array element of array in response should contain property:    [data][relationships][abstract-product-image-sets][data]    id
+    And Response body parameter should not be EMPTY:    [included]
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:73]: no need

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+
+Get_abstract_product_with_3_concretes_with_include_abstract_product_image_sets
+    When I send a GET request:    /abstract-products/${abstract_available_product_with_3_concretes}?include=abstract-product-image-sets
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response should contain the array of a certain size:    [included]    1
+    And Response should contain the array of a certain size:    [data][relationships]    1
+    And Response body parameter should not be EMPTY:    [data][relationships]
+    And Response body parameter should not be EMPTY:    [data][relationships][abstract-product-image-sets]
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:70]: no need in 69-70 lines
we don't check empty arrays if we check properties and values

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot (hunk 5 above)
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:55]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot (hunk 10 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:46]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot (hunk 11 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:73]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot (hunk 12 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:70]: Done

File: tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot (hunk 9 above)
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/mp_b2b/glue/abstract_product_endpoints/abstract_product_image_sets/positive.robot:60]: Please add check for [include][0][id] and [relationships][id]

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**